### PR TITLE
I18n console

### PR DESCRIPTION
**Pull request by RobDolares**
Related issue: https://github.com/stolostron/backlog/issues/23473

### CODE REVIEW

File: frontend/src/routes/Home/Search/components/SavedSearchQueries.tsx
Comment (KevinFCormier) [inline frontend/src/routes/Home/Search/components/SavedSearchQueries.tsx:169]: Are these predefined searches, or are they user supplied? Generally we should only provide literal strings to the `t` function; otherwise, all possible calls to the `t` function should be enumerated in a comment. Same comment for line below.

File: frontend/src/routes/Home/Search/searchDefinitions.test.ts
Comment (KevinFCormier) [inline frontend/src/routes/Home/Search/searchDefinitions.test.ts:15]: These look like leftover changes from something you tried and abandoned.

File: frontend/src/routes/Home/Search/searchDefinitions.tsx
Comment (KevinFCormier) [inline frontend/src/routes/Home/Search/searchDefinitions.tsx:1378]: I don't think there is any value in factoring this out. The uppercase function name is strange - looks like a component, but not really being used as one.

File: frontend/src/routes/Home/Search/searchDefinitions.tsx
Comment (KevinFCormier) [inline frontend/src/routes/Home/Search/searchDefinitions.tsx:1509]: Have you investigated https://github.com/open-cluster-management/backlog/issues/9184 (mentioned in the comment you removed) and whether translating this string would cause an issue?

File: frontend/src/routes/Home/Search/searchDefinitions.tsx
Comment (KevinFCormier) [inline frontend/src/routes/Home/Search/searchDefinitions.tsx:1514]: Have you investigated https://github.com/open-cluster-management/backlog/issues/9184 and whether translating this string would cause an issue?

Comment (KevinFCormier) [reply frontend/src/routes/Home/Search/searchDefinitions.tsx:1514]: Also, you only removed the comment but did not add translation...

File: frontend/src/ui-components/AcmTable/AcmTable.tsx
Comment (KevinFCormier) [inline frontend/src/ui-components/AcmTable/AcmTable.tsx:1079]: I'm not sure if this will work - we should ask the translation team if a double translation like this is OK. Some languages might not use the plural form of the word when substituting in. We might have to require that tables either have an empty state or have 2 properties to specify these strings that are used for the default empty state.

File: frontend/src/wizards/Argo/ArgoWizard.tsx
Comment (KevinFCormier) [inline frontend/src/wizards/Argo/ArgoWizard.tsx:618]: Not sure that these should be translated, since they appear to be the actual values for the select options. The actual values will be substituted into YAML, yes?

Note that the options can be an array of Option, rather than just a string, which will allow you to handle translation properly.

File: frontend/src/wizards/Argo/ArgoWizard.tsx
Comment (KevinFCormier) [inline frontend/src/wizards/Argo/ArgoWizard.tsx:690]: Careful here as well to distinguish values vs. displayed values

File: frontend/src/wizards/Placement/MatchExpression.tsx
Comment (KevinFCormier) [inline frontend/src/wizards/Placement/MatchExpression.tsx:48]: Here is an example of how to distinguish label and value.

File: frontend/src/wizards/Policy/PolicyWizard.tsx
Comment (KevinFCormier) [inline frontend/src/wizards/Policy/PolicyWizard.tsx:496]: Careful!

File: frontend/src/wizards/Placement/PlacementBinding.tsx
Comment (KevinFCormier) [inline frontend/src/wizards/Placement/PlacementBinding.tsx:117]: Another case where we need to distinguish value/label. But in this case, we are referring to a "kind" which should probably not be translated anyway. Translation team should know not to translate something like "PolicySet"... hard to say how they would know not to translate "Policy". In any case, never rely on a translation for proper functioning of the code though.

**Review by KevinFCormier — CHANGES_REQUESTED** [review]
Looks pretty good overall. For the moment, I have not reviewed for completeness. Let's get this PR wrapped up, and then we can search for any missed strings.

But there are a number of issues to take a look at. I didn't necessarily find all cases, but please have another review, especially the places where you translated select options.

File: frontend/src/routes/Home/Search/SearchResults/SearchResults.tsx
Comment (RobDolares) [reply frontend/src/routes/Home/Search/SearchResults/SearchResults.tsx:105]: This is hint for tableheader 'scope' in the result table for kind:policyreport

File: frontend/src/routes/Home/Search/SearchResults/SearchResults.tsx
Comment (KevinFCormier) [reply frontend/src/routes/Home/Search/SearchResults/SearchResults.tsx:47]: And also via ``t(`${item.tooltip}`)`` I guess

Comment (RobDolares) [reply frontend/src/routes/Home/Search/SearchResults/SearchResults.tsx:47]: yes, will update to say SearchResults table headers and applicable tooltips.

File: frontend/src/routes/Home/Search/components/SavedSearchQueries.tsx
Comment (RobDolares) [reply frontend/src/routes/Home/Search/components/SavedSearchQueries.tsx:169]: I handled this elsewhere for the suggestedQueryTemplates, so removing here.

File: frontend/src/routes/Home/Search/searchDefinitions.tsx
Comment (RobDolares) [reply frontend/src/routes/Home/Search/searchDefinitions.tsx:1378]: factored out because i wasn't able to get useTranslation hook to work within cell(), but also just refactored slightly so switch statement is also factored out into what im labeling as `GetViolations` now.

Comment (KevinFCormier) [reply frontend/src/routes/Home/Search/searchDefinitions.tsx:1378]: You could set up `searchDefinitions.tsx` to be a hook just like `useWizardStrings`... that way you also would not need to list all the terms in a big comment. This is much less error-prone, because developers will not know to update the big comment when they add search definitions.

File: frontend/src/wizards/Policy/PolicyWizard.tsx
Comment (RobDolares) [reply frontend/src/wizards/Policy/PolicyWizard.tsx:496]: updated with label/value

File: frontend/src/wizards/Placement/PlacementBinding.tsx
Comment (RobDolares) [reply frontend/src/wizards/Placement/PlacementBinding.tsx:117]: updated to [`{label:'t(...)', value: '...'}`]
Translation team said to wrap anything questionable wrt kube resources, and they will use their discretion.

File: frontend/src/wizards/Argo/ArgoWizard.tsx
Comment (RobDolares) [reply frontend/src/wizards/Argo/ArgoWizard.tsx:618]: updated to `[{label:'t(...)', value: '...'}]`

File: frontend/src/ui-components/AcmTable/AcmTable.tsx
Comment (RobDolares) [reply frontend/src/ui-components/AcmTable/AcmTable.tsx:1079]: Will first make initial inquiry about the double translation to translation team.

File: frontend/src/wizards/Argo/ArgoWizard.tsx
Comment (RobDolares) [reply frontend/src/wizards/Argo/ArgoWizard.tsx:694]: will revert to untranslated strings for now

Comment (RobDolares) [reply frontend/src/wizards/Argo/ArgoWizard.tsx:694]: reverting

File: frontend/src/ui-components/AcmTable/AcmTable.tsx
Comment (RobDolares) [reply frontend/src/ui-components/AcmTable/AcmTable.tsx:1079]: reached out to translation team 11/7/22.  Waiting for feedback

File: frontend/src/routes/Home/Search/searchDefinitions.tsx
Comment (RobDolares) [reply frontend/src/routes/Home/Search/searchDefinitions.tsx]: reverted functions until revisit changing to hook

File: frontend/public/locales/en/translation.json
Comment (KevinFCormier) [inline frontend/public/locales/en/translation.json:25]: Remove please!

File: frontend/public/locales/en/translation.json
Comment (KevinFCormier) [inline frontend/public/locales/en/translation.json:1198]: Please try to remove strings that are not actually used.

File: frontend/src/wizards/Argo/ArgoWizard.tsx
Comment (KevinFCormier) [reply frontend/src/wizards/Argo/ArgoWizard.tsx:694]: If you reverted, the change has not been pushed. Marking as unresolved.

File: frontend/src/wizards/Placement/PlacementBinding.tsx
Comment (KevinFCormier) [inline frontend/src/wizards/Placement/PlacementBinding.tsx:75]: You know what to do.

File: frontend/src/wizards/Argo/ArgoWizard.tsx
Comment (RobDolares) [reply frontend/src/wizards/Argo/ArgoWizard.tsx]: up now

File: frontend/src/ui-components/AcmTable/AcmTable.tsx
Comment (rh-tchuang) [reply frontend/src/ui-components/AcmTable/AcmTable.tsx:1079]: @KevinFCormier understood.
I'd like to double confirm that when translating the strings, if there's a segment containing possible plural form, for example "selected %s node", will there be a "selected %s nodes" for the plural candidate?

Comment (KevinFCormier) [reply frontend/src/ui-components/AcmTable/AcmTable.tsx:1079]: Hey @rh-tchuang,

Yes, with i18next, when we have an interpolation in the message with the name `count`, we automatically get a 2nd key with the suffix `_plural`. See https://github.com/stolostron/console/blob/356ab3059f1b222fee26ab8f7ee83cceb2740f31/frontend/public/locales/en/translation.json#L19:L20 for example.

The translation library automatically selects the right one at runtime based on the value of `count`. Note that this format for plurals is outdated (format v3), but for compatibility reasons with OpenShift console, we will be maintaining it for awhile. Eventually we will need to migrate to the updated format, which uses different suffixes like `_one` or `_other`. https://www.i18next.com/translation-function/plurals 

But it should be clear from the English string whether it is a plural or singular case.

The example in this PR was not really about plurals. 
`message={t('You do not have any {{items}} yet', { items: props.plural })}`
Here, `props.plural` contains a translated term for the type of item we are displaying. For example, for a table of applications, `props.plural` will be "applications". If there are no applications, the message "You do not have any applications yet" will be displayed. This means we are inserting a translated word into another translated message, which I imagine is not ideal. There is no singular form for this case.

**Pull request merged**
